pageserver/src/walreceiver.rs

```diff
@@ -118,7 +128,14 @@ fn thread_main(conf: &'static PageServerConf, tenant_id: ZTenantId, timeline_id:
     info!("WAL receiver thread started");
 
     // Look up the current WAL producer address
-    let wal_producer_connstr = get_wal_producer_connstr(tenant_id, timeline_id);
+    let wal_producer_connstr = get_wal_receiver_entry(tenant_id, timeline_id)
+        .unwrap_or_else(|| {
+            panic!(
```

Comment:
Let's change this into an Err return. (Yeah, it was a panic before, via unwrap(), but a softer error seems more appropriate here.)

Something like `.ok_or_else(anyhow!("..."))?` would work I think

Reply:
updated! The function doesn't return an error, so I need to change the code differently with `return`